test/tests/functional/pbs_dup_acc_log_for_resv.py

```diff
@@ -52,50 +52,45 @@ def setUp(self):
+    def check_log_resv_time(self, accounting_log_msg, start_tup):
         """
-        This is a helper function to reterive start and end time
-        of standing rervation from accounting log.
+        This is a helper function to retrieve start and end time
+        of standing reservation from accounting log.
         """
-        loglist_value = accounting_log_msg[start_tup][end_tup]
+        loglist_value = accounting_log_msg[start_tup][1]
         log_value = loglist_value.split()
-        if check_status == "start":
+        actual_reservation_time = []
+        check_status = ['start', 'end']
+        for status in check_status:
             for attribute in log_value:
                 attribute_value = attribute.split("=")
-                if attribute_value[0] == 'start':
-                    actual_reservation_time = attribute_value[1]
-        elif check_status == "end":
-            for attribute in log_value:
-                attribute_value = attribute.split("=")
-                if attribute_value[0] == 'end':
-                    actual_reservation_time = attribute_value[1]
-        return int(actual_reservation_time)
+                if attribute_value[0] == status:
+                    actual_reservation_time.append(int(attribute_value[1]))
+
+        return actual_reservation_time
 
     def differentiate_resv_instance(self, expected_start, expected_end,
                                     got_start, got_end, reserve_index):
         """
-        This function to differentiate log match
-        for each instance of standing reservation
-        on basis of comparing expected and got start and end time
-        of standing reservation of each instance(reserve_index)
+        This is a helper function to differentiate log match for each
+        instance(reserve_index) of standing reservation.
+        As input function accept expected_start,expected_end,got _start,
+        got_end time value and reserve_ index.
+        By comparison of expected_start & got_start and expected_end &
+        got_end time value,function will log info for specific
+        instance(reserve_index) of standing reservation.
         """
 
         self.assertTrue(((expected_start == got_start) and
                          (expected_end == got_end)),
                         'Time in the accounting log does not match with'
                         'the expected time for the standing reservation')
-        if reserve_index == '1':
-            self.logger.info(
-                'Matched accounting log for reserve_index instance1')
-        elif reserve_index == '2':
-            self.logger.info(
-                'Matched accounting log for reserve_index instance2')
-        elif reserve_index == '3':
-            self.logger.info(
-                'Matched  accounting log for reserve_index instance3')
+
+        match_string = 'Matched accounting log for instance' + reserve_index
+        if reserve_index in ('1', '2', '3'):
```

Comment:
My understanding is that if we expand the testsuite to check for 4th or 5th instance of a standing reservation, we would need to modify this function to include them. Also, this function should not restrict the caller to only first three instances of the reservation. We can remove this check...